src/Square.php

```diff
@@ -0,0 +1,28 @@
+<?php
+namespace Shape;
+
+class Square implements ShapeInterface
+{
+    protected float $length;
+    private string $name;
+
+    public function __construct(float $length)
+    {
+        $this->length = $length;
+    }
+
+
+    public function setName(string $name) {
```

Comment:
Please do not forget about return type declarations.
P.S. There is also the one which is specified when there actually no return value.

Reply:
As name is required for each shape and object will not be valid when name is not specified, also calling "getName" method will return error in case you'll not call "setName" before that,could it be a better way how to assign name for a shape?